src/arkode/arkode_arkstep.c

```diff
@@ -3715,8 +3760,7 @@ int ARKStepCreateAdjointStepper(void* arkode_mem, N_Vector sf,
 
   /* SUNAdjointStepper will own the SUNSteppers and destroy them */
   errcode = SUNAdjointStepper_Create(fwd_stepper, SUNTRUE, adj_stepper, SUNTRUE,
-                                     nst - 1, sf, ark_mem->tretlast,
-                                     ark_mem->checkpoint_scheme,
+                                     nst - 1, sf, tf, ark_mem->checkpoint_scheme,
```

Comment:
Swap `tf` and `sf` to match the time then vector order of other constructors. Do we ensure `nst - 1` and `tf` coincide? 
```suggestion
                                     nst - 1, tf, sf, ark_mem->checkpoint_scheme,
```